File: packages/agentmark/src/prompts/image.ts
import { Adapter, TemplateEngine, JSONObject, PromptMetadata, ImageConfig } from "../types";

export class ImagePrompt<
  T extends { [K in keyof T]: { input: any; output: any } },
  A extends Adapter<T>,
  K extends keyof T & string
> {
  protected templateEngine: TemplateEngine;
  protected adapter: A;
  public path: K;
  public template: unknown;
  
  constructor(template: unknown, templateEngine: TemplateEngine, adapter: A, path: K) {
    this.template = template;
    this.templateEngine = templateEngine;
    this.adapter = adapter;
    this.path = path;
  }

  async format(
    props: T[K]["input"],
    options: JSONObject = {}
  ): Promise<ReturnType<A['adaptImage']>> {
    const compiledTemplate = await this.templateEngine.compile(this.template, props) as ImageConfig;
    const metadata: PromptMetadata = { props, path: this.path, template: this.template };
    return this.adapter.adaptImage(compiledTemplate, options, metadata);
  }
} 

File: packages/prompt-core-python/src/agentmark/prompt_core/prompts/image.py
"""Image prompt class."""

from typing import Any

from ..schemas import ImageConfigSchema
from .base import BasePrompt


class ImagePrompt(BasePrompt[ImageConfigSchema]):
    """Image generation prompt."""

    async def format(
        self,
        props: dict[str, Any] | None = None,
        **options: Any,
    ) -> Any:
        """Format the image prompt.

        Args:
            props: Props to pass to the template
            **options: Additional adapter options

        Returns:
            Adapted image prompt output
        """
        compiled = await self._compile(props)
        adapt_options = self._build_adapt_options(options)
        return self._adapter.adapt_image(compiled, adapt_options)
